### PR TITLE
Workaround for missing isopen check in FDWatcher

### PR DESCRIPTION
**Pull request by vtjnash**
(possible multithread race with this still needs to be fixed)

### CODE REVIEW

**Comment by StefanKarpinski** [comment]
macOS tests timed out so I restarted them.

File: src/Curl/Multi.jl [hunk 1]
```diff
@@ -150,16 +150,17 @@ function socket_callback(
...
             catch err
-                err isa EOFError && break
+                err isa EOFError && return
                 rethrow()
             end
             flags = CURL_CSELECT_IN  * isreadable(events) +
                     CURL_CSELECT_OUT * iswritable(events) +
                     CURL_CSELECT_ERR * events.disconnect
             lock(multi.lock) do
+                watcher.readable || watcher.writable || return # !isopen
```
Comment (StefanKarpinski) [inline src/Curl/Multi.jl:163]: Should `watcher.disconnect` be in here as well or just these two?

Comment (vtjnash) [reply src/Curl/Multi.jl:163]: There is no disconnect field

**Comment by vtjnash** [comment]
Are we pretty sure https://github.com/JuliaLang/Downloads.jl/runs/4083466739?check_suite_focus=true was not a bug here?

File: src/Curl/Multi.jl (hunk 1 above)
Comment (StefanKarpinski) [reply src/Curl/Multi.jl:163]: Ah, I was thinking of the `events` object.

**Comment by StefanKarpinski** [comment]
The root error was this:
```
RequestError("https://untrusted-root.badssl.com", 28, "Connection timed out after 75933 milliseconds", Response(nothing, "https://untrusted-root.badssl.com/", 0, "", Pair{String,String}[]))
```
I guess it could be a bug but it seems more likely that the request either got lost or the server didn't reply in time.

**Comment by StefanKarpinski** [comment]
Merging since this is an improvement in any case.

**Pull request merged**